app/controllers/admin/custom_fields/hierarchy/items_controller.rb

```diff
@@ -73,45 +85,43 @@ def destroy
             .new
             .delete_branch(item: @hierarchy_item)
             .either(
-              ->(_) { update_via_turbo_stream(component: ItemsComponent.new(custom_field: @custom_field)) },
+              ->(_) { update_via_turbo_stream(component: ItemsComponent.new(item: @hierarchy_item.parent)) },
               ->(errors) { update_flash_message_via_turbo_stream(message: errors.full_messages, scheme: :danger) }
             )
 
           respond_with_turbo_streams
         end
 
         def deletion_dialog
-          respond_with_dialog DeleteItemDialogComponent.new(custom_field: @custom_field,
-                                                            hierarchy_item: @hierarchy_item)
+          respond_with_dialog DeleteItemDialogComponent.new(custom_field: @custom_field, hierarchy_item: @hierarchy_item)
         end
 
         private
 
         def item_input
-          input = { parent: @custom_field.hierarchy_root, label: params[:label] }
-          input[:short] = params[:short] unless params[:short].empty?
-
-          input
+          { parent: @hierarchy_item, label: params[:label], short: params[:short] }
         end
 
         def add_errors_to_form(validation_result)
           validation_result.errors(full: true).to_h.each do |attribute, errors|
-            @custom_field.errors.add(attribute, errors.join(", "))
+            @hierarchy_item.errors.add(attribute, errors.join(", "))
           end
 
           new_item_form_data = { show: true, label: validation_result[:label], short: validation_result[:short] }
-          update_via_turbo_stream(component: ItemsComponent.new(custom_field: @custom_field, new_item_form_data:),
-                                  status: :unprocessable_entity)
+          update_via_turbo_stream(
+            component: ItemsComponent.new(item: @hierarchy_item, new_item_form_data:), status: :unprocessable_entity
+          )
         end
 
-        def find_model_object(object_id = :custom_field_id)
-          super
+        def find_model_object
```

Comment:
🟡 if we do not call super, lets maybe rename the method to `find_custom_field`.

Reply:
This leads to other errors. This dude here does a bunch of stuff for the sidebars etc.